wsmux/session.go

```diff
@@ -0,0 +1,221 @@
+package wsmux
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"net"
+	"os"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/taskcluster/slugid-go/slugid"
+)
+
+const (
+	defaultQueueSize       = 10
+	defaultStreamQueueSize = 10
+)
+
+var (
+	errRemoteClosed  = fmt.Errorf("remote session is no longer accepting connections")
+	errAcceptTimeout = fmt.Errorf("accept timed out")
+	errBrokenPipe    = fmt.Errorf("broken pipe")
+)
+
+// Session ...
+type Session struct {
+	streamLock sync.RWMutex
+
+	streams    map[uint32]*Stream
+	writes     chan frame
+	streamChan chan *Stream
+
+	conn     *websocket.Conn
+	connLock sync.Mutex
+
+	remoteClosed bool
+	closed       bool
+	nextID       uint32
+
+	onRemoteClose  func(*Session)
+	acceptDeadline time.Duration
+	logger         *log.Logger
+}
+
+func newSession(conn *websocket.Conn, server uint32, onRemoteClose func(*Session)) *Session {
```

Comment:
let's make it a boolean, and then add a comment explaining why `nextID` is either `1` or `0` to start with...